File: estructuras_de_datos.js
/*
arrays
objetos
sets
*/

var miArray=[1,28,3,7,12]
console.log(typeof(miArray))//object

//cantidad de elementos
console.log(miArray.length)//propiedad
console.log(miArray.pop())//12
console.log(miArray)
console.log(miArray.reverse())

//push
miArray.push("f")
console.log(miArray)

miArray.shift()//aplicar el metodo
console.log(miArray)//resultado
miArray.unshift("b")
console.log(miArray)
console.log(miArray.shift())//b
//ver que retorna la aplicacion del metodo
console.log(miArray.unshift("e"))//5

//Objeto Javascript
var estudiantes={
    nombre:"andres",
    edad:21
}
console.log(typeof(estudiantes))//object
console.log(estudiantes.edad)

estudiantes.ciudad="palmira"
console.log(estudiantes)

var datos=Array(28,50,60)
console.log(datos)

var conjunto=new Set()//constructor
console.log(conjunto)
conjunto.add(8)
console.log(conjunto)

File: estructuras_de_datos.py
'''
listas
tuplas
diccionarios
sets
'''

mi_lista=[1,28,3,7,12]
print(type(mi_lista))#<class 'list'>

#cantidad de elementos
print(len(mi_lista))
#uso de pop
print(mi_lista.pop())#12
print(mi_lista)

#reverse
print(mi_lista.reverse())
print(mi_lista)

#append
print(mi_lista.append("f"))
print(mi_lista)

#Dicionarios
estudiantes = {
    "nombre":"andres",
    "edad":21
}
print(type(estudiantes))#<class 'dict'>

#como acceder a un valor
print(estudiantes.get("edad"))
print(estudiantes["edad"])#variable 2 de acceder

#agregar un nuevo par clave valor
estudiantes["ciudad"]="palmira"
print(estudiantes)

mi_conjunto=set()#Constructor de conjuntos
mi_conjunto.add(8)
print(mi_conjunto)